pkg/edvprovider/couchdbedvprovider/couchdbedvprovider.go

```diff
@@ -137,7 +132,146 @@ func (c CouchDBEDVStore) Query(query *models.Query) ([]string, error) {
 	return c.filterDocsByQuery(idsOfDocsWithMatchingQueryIndexName, query)
 }
 
-func (c CouchDBEDVStore) findDocsMatchingQueryIndexName(queryIndexName string) (map[string]struct{}, error) {
+func (c *CouchDBEDVStore) checkNewDocumentForConflictingIndexWithAnyExistingDocs(
```

Comment:
This name is too long. How about 'validateDoc'? Other funcs: validateAttributes, etc.

Reply:
Done. Let me know if you think the new names are ok. I tried to make them as short as possible without ambiguity.